src/collections.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Handy generic functions for operating on JS collections.
+ *
+ * Any function here is the sort of thing that perhaps should be in the
+ * language's standard library.
+ *
+ * See also
+ *  * `jsBackport.js`, for things that actually are in newer versions of JS,
+ *    or late-stage proposals for future versions
+ *  * `generics.js`, for things that operate on types
+ *
+ * @flow strict-local
+ */
+
+/**
+ * Like `Array#map`, but returns null if `f` ever returns a nullish value.
+ *
+ * If `f` returns nullish at any element, the iteration stops and `f` is not
+ * called on any later elements.
+ *
+ * This replaces either (a) writing an explicit loop, in order to be able to
+ * abort on a nullish value, or (b) using `Array#map` and then running
+ * through the list again to look for nullish values.
+ */
+export function mapOrNull<T, U>(
+  ths: $ReadOnlyArray<T>,
```

Comment:
Hmm, what does `ths` stand for? After reading the comment about `thisArg`, it makes me think it might have to do with that, but I think it doesn't.

Reply:
Ah, I should make it clearer, then. :slightly_smiling_face:

It's meant to stand for "this" -- as in, if you imagine this *were* in the stdlib it'd be a method on Array, so the array it's mapping over would be `this`. But of course `this` is a keyword and I can't just call the parameter that, so I have to munge the name.

In Python there's a fairly common pattern that calls for a function to take an argument that is a class and you'd like to call it "class", and similarly `class` is a keyword and you can't. The standard solution there -- [here it is](https://docs.python.org/3/library/functions.html#classmethod) right in the language reference docs -- is to call it `cls`. So I think in my head I was echoing that usage. But that's not so helpful for a reader who hasn't spent time writing Python :wink: 

Given the "thisArg" thing, probably anything that tries to say "this" will be confusing unless separately explained. Maybe I'll call it `array`.